app/services/xccdf_report_parser.rb

```diff
@@ -46,36 +44,46 @@ def inventory_api
   end
 
   def save_host
-    @host = Host.find_or_initialize_by(
-      id: @host_inventory_id,
-      account_id: @account.id
-    )
-    @host.update(name: report_host) unless @host.name == report_host
-    new_profiles = host_new_profiles
-    @host.profiles << new_profiles if new_profiles.present?
+    @host = Host.find_or_initialize_by(id: @host_inventory_id,
+                                       account_id: @account.id)
+    @host.update(name: report_host, profiles: host_profiles)
     inventory_api.sync
+    invalidate_cache
+  end
+
+  def host_profiles
+    host_profile = @profiles.find do |profile|
+      profile.ref_id == @test_result_file.test_result.profile_id
+    end
+
+    unless @host.profiles.pluck(:ref_id).include?(host_profile.ref_id)
+      host_profile = host_profile.dup
```

Comment:
@dLobatog I'd like some help figuring this out. At this point in the import, we've imported all the benchmark info, including profiles, and those profiles have `nil` `account_id`. The problem is that all the associated rules (via `ProfileRules` table) are not duplicated, so this dupe profile has no rules. I'm leaning towards adding `parent_id` now like we talked about and only ever have profiles with `nil` accound_id be associated with rules. What do you think?

Reply:
I think for now, I'm just going to copy all the rules from the parent profile without adding `parent_id`